### ISSUE TITLE
inference failed

### ISSUE THREAD

**Issue by pgrossomoreira** [blace-ai/blace-ai#9]
I have a proprietary model provided by a 3rd party vendor that I'd like to run using blaceai. It's a motion vector generation model and it comes as a `.pt` file.

In order to get this model running with blaceai, I had to write a `.json` file to pass to `model-wizard` which generates the C++ header and payload that blaceai will consume when I run my C++ program.

When I run my program, I observe the following error messages:
```
inference failed on first try...
Unknown exception
failed on second attempt too...
Unknown exception
Running of the ai model failed. This is probably due to insufficient amount of RAM or VRAM.
```

I am at a loss on what I need to do to get this working as the "Unknown exception" message doesn't really give any hints. I will describe my process and I'd appreciate some help with figuring out where something went wrong.

### 1. Inspect the model

I used the following python script to try and understand the model's inputs and outputs:

```
import torch

model = torch.jit.load("NNFlowVector_A.pt")

print('Inputs:')
for input in model.graph.inputs():
    if not input.debugName().startswith('self.'):
        print(f'- {input.debugName()} ({input.type()})')

print('Outputs:')
for output in model.graph.outputs():
    print(f'- {output.debugName()} ({output.type()})')

cpu = torch.device("cpu")
model.to(cpu)

mkinput = lambda: torch.randn(1, 3, 640, 640, dtype=torch.float32)

example_input1 = mkinput()
example_input2 = mkinput()
iters = torch.tensor(1)

output = model(example_input1, example_input2, iters)

print("Example Output Shape:", output.shape if hasattr(output, 'shape') else type(output))
```

This gives the following output:
```
Inputs:
- image1.1 (Tensor)
- image2.1 (Tensor)
- iters.1 (Tensor)
Outputs:
- 116 (Tensor)
Example Output Shape: torch.Size([1, 2, 640, 640])
```

From which I conclude that the model has three inputs (two images and one integer) and one output (an image with 2 channels)

### 2. Write blaceai's JSON model description

This is what I came up with:

```
{
    "inputs": [
       {
            "name": "image1.1",
            "content": {
                "metaTensor": {
                    "data_type": "FLOAT_32",
                    "normalization": {
                        "norm": "ZERO_TO_ONE",
                        "allow_overflow": true
                    },
                    "color_format": "RGB",
                    "order": "BCHW"
                }
           },
           "inputSizes": [
                { "modulo_by": 1, "min": 1, "max": 1 },
                { "modulo_by": 1, "min": 1, "max": 4 },
                { "modulo_by": 4, "min": 8,  "max": 6000 },
                { "modulo_by": 4, "min": 8, "max": 6000 }
           ]
       },
       {
            "name": "image2.1",
            "content": {
                "metaTensor": {
                    "data_type": "FLOAT_32",
                    "normalization": {
                        "norm": "ZERO_TO_ONE",
                        "allow_overflow": true
                    },
                    "color_format": "RGB",
                    "order": "BCHW"
                }
           },
           "inputSizes": [
                { "modulo_by": 1, "min": 1, "max": 1 },
                { "modulo_by": 1, "min": 1, "max": 4 },
                { "modulo_by": 4, "min": 8,  "max": 6000 },
                { "modulo_by": 4, "min": 8, "max": 6000 }
           ]
       },
       {
            "name": "iters.1",
            "content": {
                "metaTensor": {
                    "data_type": "INT_32"
                }
           },
           "inputSizes": [
                {  "modulo_by": 1,  "min": 1, "max": 10 }
           ]
       }
    ],
    "outputs": [
        {
            "name": "116",
            "content": {
                "metaTensor": {
                    "data_type": "FLOAT_32",
                    "normalization": {
                        "norm": "UNKNOWN_VALUE_RANGE",
                        "allow_overflow": true
                    },
                    "color_format": "UV",
                    "order": "BCHW"
                }
           },
           "outputSizes": [
                {
                    "output_which_input": 0,
                    "output_which_index": 0
                },
                {
                    "output_fixed_size": 2
                },
                {
                    "output_which_input": 0,
                    "output_which_index": 2
                },
                {
                    "output_which_input": 0,
                    "output_which_index": 3
                }
           ]
       }
    ],
    "torchscriptModel": {
        "can_use_fp16": false
    }
}
```

and then run

`model-wizard -m NNFlowVector_A.pt -j NNFlowVector_A.json -o nnflow`

### 3. Run with blaceai

I don't have a minimal reproducible example using pure C++ and blaceai only, but I can describe what my C++ program does. I basically just modified existing code that I had used to successfully run RAFT and VideoFlow:
- I load the model with `blace::util::registerModel()`
- I create two `blace::ops::FromRawMemoryOp`, one for each image input
- I create one `blace::ops::FromIntOp` for the third input of the model (number of iterations)
- I create one `blace::ops::InferenceOp` and pass the three previous operators to the inputs list
- I create an `blace::computation_graph::GraphEvaluator` with the inference operator as argument
- I invoke `evaluateToTorchTensor()` on that evaluator

The result is the error messages I showed earlier.

How can I figure out what's wrong?

Thank you.

Kind regards,
-Pedro Moreira

**Comment by blace-ai-transfer** [comment]
Hey Pedro,
soon we release a new version with better error logs. Something I can already tell you is that the third input needs to be a tensor (with integer datatype) and not a FromInt op. The FromInt op constructs simple integer objects that are passed to the model without being converted to a tensor.

You could try to create your integer like:

```
std::vector<int32_t> i = {4}; // 4 iterations

blace::RawMemoryObject int_input(i.data_ptr(), ml_core::DataTypeEnum::INT_32,
                           ml_core::ARBITRARY_CHANNELS, {1}, ml_core::C,
                           ml_core::UNKNOWN_VALUE_RANGE,
                           ml_core::BlaceHash(i[0], 0, 0, 0), false);
```
